tests/unit/s2n_npn_extension_test.c

```diff
@@ -149,7 +149,13 @@ int main(int argc, char **argv)
             EXPECT_BYTEARRAY_EQUAL(s2n_get_application_protocol(client_conn), protocols[0], strlen(protocols[0]));
         }
 
-        /* No match exists */
+        /*
+         *= https://datatracker.ietf.org/doc/id/draft-agl-tls-nextprotoneg-04#section-4
+         *= type=test
+         *# In the event that the client doesn't support any of server's protocols, or
+         *# the server doesn't advertise any, it SHOULD select the first protocol
```

Comment:
Should we also cover the "server doesn't advertise any" case?

Reply:
Hmmm actually we currently error when this happens. Do you think we should be erroring? I don't know why the server would send an empty extension though...

Reply:
I don't know why either, but it sounds like it's allowed, so I don't think we should error.